src/ProductBundle/Controller/ProductController.php

```diff
@@ -276,4 +279,12 @@ public function variationToProductAction($productId, $slug)
 
         return $response;
     }
+
+    /**
+     * @return Request
+     */
+    private function getCurrentRequest()
```

Comment:
can we add this to the CRUDController? Or create an BaseEcommerceCRUDController to avoid this duplication?

Reply:
mb use trait?

Reply:
sounds good 👍 

WDYT @greg0ire ?

Reply:
It's something that will be removed so a trait kinda makes sense here, because it's arguably easier to remove, and allows to keep that method private.

Reply:
we should add a `NEXT_MAJOR` above the usage.

What about a deprecation in the trait when it will be removed?

Reply:
I made a try.
We'll remove `CurrentRequestTrait` in 3.x and inject Request $request in actions parameters.